libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateInput.tsx

```diff
@@ -125,95 +124,118 @@ const useDateInput = ({
+            onClick,
             onChange,
+            dateFormat,
+            accessibilityConfig,
+            isMobile,
+            isTimeEnabled,
             isDateOrderError,
             errorMessageTexts,
         },
-    );
-
-    const inputLabelId = useId();
-    const inputErrorId = useId();
-
-    const MobileInput = (
-        <DateRangePickerInputFieldBody
-            type="date"
-            className={cx(
-                "s-date-range-picker-date",
-                "gd-date-range-picker-input",
-                "gd-date-range-picker-input-native",
-                {
-                    "gd-date-range-picker-input-error": errorMessage,
-                    "has-error": errorMessage,
-                },
-            )}
-            onChange={onMobileDateChange}
-            value={getPlatformStringFromDate(value)}
-        />
-    );
-
-    const DesktopInput = (
-        <div
-            className={cx("gd-date-range-picker-input", {
-                "gd-date-range-picker-input-error": !!errorMessage,
-                "has-error": !!errorMessage,
-            })}
-        >
-            <span className="gd-icon-calendar" />
-            <input
-                onKeyDown={onKeyDown}
-                ref={ref}
-                aria-label={accessibilityConfig.ariaLabel}
-                placeholder={dateFormat}
-                onChange={onDateInputChange}
-                onClick={onClick}
-                onBlur={onDateInputBlur}
-                value={inputValue}
-                className="input-text s-date-range-picker-input-field"
-                aria-labelledby={inputLabelId}
-                aria-describedby={buildAriaDescribedByValue([
-                    accessibilityConfig.inputHintId,
-                    errorMessage ? inputErrorId : undefined,
-                ])}
-                {...(errorMessage ? { "aria-invalid": true } : {})}
+        ref,
+    ) => {
+        const {
+            inputValue,
+            errorMessage,
+            onDateInputChange,
+            onDateInputKeyDown,
+            onMobileDateChange,
+            onDateInputBlur,
+        } = useDateInput({
+            value,
+            dateFormat,
+            onChange,
+            onKeyDown,
+            isDateOrderError,
+            errorMessageTexts,
+        });
+
+        const inputLabelId = useId();
+        const inputErrorId = useId();
+
+        const MobileInput = (
+            <DateRangePickerInputFieldBody
+                type="date"
+                className={cx(
+                    "s-date-range-picker-date",
+                    "gd-date-range-picker-input",
+                    "gd-date-range-picker-input-native",
+                    {
+                        "gd-date-range-picker-input-error": errorMessage,
+                        "has-error": errorMessage,
+                    },
+                )}
+                onChange={onMobileDateChange}
+                value={getPlatformStringFromDate(value)}
```

Comment:
Believe current aria options for the error are missing on mobile, but if is not part of the ticket scope, feel free to ignore it.